pkg/cluster/config_test.go

```diff
@@ -124,9 +124,12 @@ func TestIsValidKubernetesVersion(t *testing.T) {
 		version      string
 		errorMessage string
 	}{
-		{"cluster version lower than supported version ", "v1.17.1", "Unsupported server version: v1.17.1. Supported version is v1.20.x"},
+		{"cluster version lower than supported version ", "v1.17.1", "Unsupported server version: " +
+			"v1.17.1. Supported versions are v1.20.x to v1.21.x"},
 		{"cluster version equals supported version", "v1.20.0", ""},
-		{"cluster version greater than supported version", "v1.21.0", "Unsupported server version: v1.21.0. Supported version is v1.20.x"},
+		{"cluster version equals supported version", "v1.21.4", ""},
+		{"cluster version greater than supported version ", "v1.22.2", "Unsupported server version: " +
+			"v1.22.2. Supported versions are v1.20.x to v1.21.x"},
```

Comment:
I dont see a reason to check the exact error message, to me all it adds is extra work when bumping the expected kube version. Instead we should check if there was an error or not, since that is what we care about. Since you are changing the function/test I think it would be a good time to make this change.